build: bring first-time greeting script in-house

Planeshifter opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-  replaces usage of the `actions/first-interaction` action, which we had frequent issues with, with our own in-house script.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves https://github.com/stdlib-js/metr-issue-tracker/issues/47.

## Questions

> Any questions for reviewers of this pull request?

I left the content of the posted comments the same for now. It's not clear to me what changes would be desirable: For example, I thought first it may be a good idea to mention slash commands on first PR, but since any workflow runs have to be approved at that point, it's actually not a good idea to point new contributors to this functionality. Similarly, I was thinking of linking to contribution and development guide, but people should have done that before raising a PR and it's probably better to have presumption of innocence instead of auto-assuming people need to be reminded of our development practices. 

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Planeshifter committed:
build: bring first-time greeting script in-house

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: passed
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented:
Re: updating comment. In offline discussions, we discussed

> Update bot comment for first PR by a new contributor with more helpful instructions and orientation

I believe the idea was to, e.g., provide a list of `make` commands with example usage for compiling native add-ons and running tests, examples, benchmarks, etc. We could also provide an overview of the review process. E.g., a reviewer will take a look. If things seem in a good enough state, a maintainer will approve GitHub workflows. If those fail, please try to remedy any failures. To reproduce build failures locally, run the following commands. We could also state that, if a maintainer has not responded within 2 weeks, please feel free to ping the "reviewers" team by leaving a comment on this PR. You can also link to the developer FAQ. Etc.

In short, I think you can probably use your imagination a bit more for what would be potentially helpful.

Planeshifter committed:
build: add missing checkout step

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
build: expand first PR and first issue comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: passed
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on .github/workflows/scripts/first_time_greeting:
```diff
@@ -0,0 +1,219 @@
+	## Getting Started
+
+	-   Please read our [contributing guidelines][stdlib-contributing] if you haven't already.
+	-   For development guidance, refer to the [development guide][stdlib-development]
```
```suggestion
	-   For development guidance, refer to the [development guide][stdlib-development].
```

kgryte commented on .github/workflows/scripts/first_time_greeting:
```diff
@@ -0,0 +1,219 @@
+	-   [Developtment Guide][stdlib-development]
+	-   [Gitter channel][stdlib-gitter]
+	-   [make rules for running examples][make-docs-examples]
+	-   [make rules for running project unit tests][make-docs-test]
```
```suggestion
	-   [make rules for running unit tests][make-docs-test]
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM. Although, again, we really should not be inlining Markdown documents in Bash scripts. It makes scripts much harder to grok, and it circumvents all the Markdown linting processes we've created.
